packages/next/src/server/base-server.ts

```diff
@@ -2141,141 +2148,138 @@ export default abstract class Server<ServerOptions extends Options = Options> {
       // served by the server.
       let result: RenderResult
 
-      if (components.routeModule?.definition.kind === RouteKind.APP_ROUTE) {
-        const routeModule = components.routeModule as AppRouteRouteModule
-
-        const context: AppRouteRouteHandlerContext = {
-          params: opts.params,
-          prerenderManifest,
-          renderOpts: {
-            // App Route's cannot postpone, so don't enable it.
-            ppr: false,
-            originalPathname: components.ComponentMod.originalPathname,
-            supportsDynamicHTML,
-            incrementalCache,
-            isRevalidate: isSSG,
-          },
-        }
+      if (routeModule) {
```

Comment:
Finally I also pulled the helper methods for type detection up and exposed them.